src/compression/mod.rs

```diff
@@ -297,7 +303,7 @@ impl Compression {
 // FIXME this should really be done inside each compression method
 
 #[allow(unused)]
-fn convert_current_to_little_endian(bytes: Bytes<'_>, channels: &ChannelList, rectangle: IntegerBounds) -> ByteVec { // TODO is this really not already somewhere else?
+fn convert_current_to_little_endian<'a>(bytes: Bytes<'a>, channels: &ChannelList, rectangle: IntegerBounds) -> Cow<'a, [u8]> { // TODO is this really not already somewhere else?
```

Comment:
While `Cow` is probably good performance-wise, I'm wondering if the code would be simpler in total if we were to change the signature to `fn convert_current_to_little_endian(bytes: &mut [u8], channels: &ChannelList, rectangle: IntegerBounds) { ... }`?